### ISSUE TITLE
Flaky Test – io.camunda.zeebe.engine.processing.processinstance.CreateProcessInstanceAnywhereTest#shouldSubscribeToScopeEvents

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda/zeebe#9557]
**Summary**

> Try to answer the following as best as possible

- How often does the test fail? once thus far
- Does it block your work? no, but we should fix it while we are working on this topic
- Do we suspect that it is a real failure? Possibly

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 
Expecting actual to contain the specified subsequence but failed to find the element at subsequence index 1 in actual:
subsequence element not found in actual:
  ("timer-boundary-event", BOUNDARY_EVENT, ELEMENT_COMPLETED)
actual:
  [("process", PROCESS, ELEMENT_ACTIVATING),
    ("process", PROCESS, ELEMENT_ACTIVATED),
    ("subprocess", SUB_PROCESS, ELEMENT_ACTIVATING),
    ("subprocess", SUB_PROCESS, ELEMENT_ACTIVATED),
    ("task", SERVICE_TASK, ACTIVATE_ELEMENT),
    ("task", SERVICE_TASK, ELEMENT_ACTIVATING),
    ("task", SERVICE_TASK, ELEMENT_ACTIVATED),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_ACTIVATING),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_ACTIVATED),
    ("message-boundary-event", BOUNDARY_EVENT, COMPLETE_ELEMENT),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_COMPLETING),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_COMPLETED),
    ("sequenceFlow_e69dac8f-c702-4d2e-bbcd-14f2f1847305", SEQUENCE_FLOW, SEQUENCE_FLOW_TAKEN),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ACTIVATE_ELEMENT),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_ACTIVATING),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_ACTIVATED),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_COMPLETING),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_COMPLETED),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_ACTIVATING),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_ACTIVATED),
    ("message-boundary-event", BOUNDARY_EVENT, COMPLETE_ELEMENT),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_COMPLETING),
    ("message-boundary-event", BOUNDARY_EVENT, ELEMENT_COMPLETED),
    ("sequenceFlow_e69dac8f-c702-4d2e-bbcd-14f2f1847305", SEQUENCE_FLOW, SEQUENCE_FLOW_TAKEN),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ACTIVATE_ELEMENT),
    ("task", SERVICE_TASK, COMPLETE_ELEMENT),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_ACTIVATING),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_ACTIVATED),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_COMPLETING),
    ("endEvent_40366a6a-d5ed-41a4-b5ca-69c48b1fab96", END_EVENT, ELEMENT_COMPLETED),
    ("task", SERVICE_TASK, ELEMENT_COMPLETING),
    ("task", SERVICE_TASK, ELEMENT_COMPLETED),
    ("sequenceFlow_3240c7af-6690-4a8c-a081-a5bcbca48861", SEQUENCE_FLOW, SEQUENCE_FLOW_TAKEN),
    ("endEvent_4b4a8c5a-3bbb-454e-8b10-73c13a5d8824", END_EVENT, ACTIVATE_ELEMENT),
    ("endEvent_4b4a8c5a-3bbb-454e-8b10-73c13a5d8824", END_EVENT, ELEMENT_ACTIVATING),
    ("endEvent_4b4a8c5a-3bbb-454e-8b10-73c13a5d8824", END_EVENT, ELEMENT_ACTIVATED),
    ("endEvent_4b4a8c5a-3bbb-454e-8b10-73c13a5d8824", END_EVENT, ELEMENT_COMPLETING),
    ("endEvent_4b4a8c5a-3bbb-454e-8b10-73c13a5d8824", END_EVENT, ELEMENT_COMPLETED),
    ("subprocess", SUB_PROCESS, COMPLETE_ELEMENT),
    ("subprocess", SUB_PROCESS, ELEMENT_COMPLETING),
    ("subprocess", SUB_PROCESS, ELEMENT_COMPLETED),
    ("sequenceFlow_dadb18b0-0ecb-41b8-a684-812d3ec049e7", SEQUENCE_FLOW, SEQUENCE_FLOW_TAKEN),
    ("endEvent_62421b46-e382-465c-9ef7-bbef8fc78aae", END_EVENT, ACTIVATE_ELEMENT),
    ("endEvent_62421b46-e382-465c-9ef7-bbef8fc78aae", END_EVENT, ELEMENT_ACTIVATING),
    ("endEvent_62421b46-e382-465c-9ef7-bbef8fc78aae", END_EVENT, ELEMENT_ACTIVATED),
    ("endEvent_62421b46-e382-465c-9ef7-bbef8fc78aae", END_EVENT, ELEMENT_COMPLETING),
    ("endEvent_62421b46-e382-465c-9ef7-bbef8fc78aae", END_EVENT, ELEMENT_COMPLETED),
    ("process", PROCESS, COMPLETE_ELEMENT),
    ("process", PROCESS, ELEMENT_COMPLETING),
    ("process", PROCESS, ELEMENT_COMPLETED)]
subsequence:
  [("message-boundary-event", BOUNDARY_EVENT, ELEMENT_COMPLETED),
    ("timer-boundary-event", BOUNDARY_EVENT, ELEMENT_COMPLETED),
    ("process", PROCESS, ELEMENT_COMPLETED)]
	at io.camunda.zeebe.engine.processing.processinstance.CreateProcessInstanceAnywhereTest.shouldSubscribeToScopeEvents(CreateProcessInstanceAnywhereTest.java:778)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:234)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:228)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:131)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
</pre>
</details>

**Hypotheses**

It seems like the message that is send gets correlated twice, messing up the assertion that waits for the timer boundary event to be triggered. Maybe there is multiple message subscriptions created?

**Comment by saig0** [comment]
*Hypotheses*

The order of the message publish command and the time travel can cause issues. First, the message is published, and then the time is increased. If the time is increased before the message is acknowledged (i.e. `message_subscription: correlate` command) then the message is correlated again because of the retry mechanics. 

If the assumption is correct, we should switch the commands, first the timer and then the message.

**Comment by saig0** [comment]
The assumption is correct. The following record stream shows the issue: 

```
C DEPLOYMENT         CREATE         - #001->  -1   -1 - 
E PROC               CREATED        - #002->#001 K001 - process.xml -> "process" (version:1)
E DEPLOYMENT         CREATED        - #003->#001 K002 - process.xml
E DEPLOYMENT         FULLY_DISTR    - #004->#001 K002 - 
C PROC_INST_CREATION CREATE         - #005->  -1   -1 - new <process "process"> with variables: {key=key-3515601627347623276}
E VAR                CREATED        - #006->#005 K004 - key->"key-3515601627347623276" in <process [K003]>
E PROC_INST          ACTIVATING     - #007->#005 K003 - PROCESS "process" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #008->#005 K003 - PROCESS "process" in <process "process"[K003]>
E PROC_INST          ACTIVATING     - #009->#005 K005 - SUB_PROCESS "subprocess" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #010->#005 K005 - SUB_PROCESS "subprocess" in <process "process"[K003]>
E PROC_MSG_SUB       CREATING       - #011->#005 K006 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E TIMER              CREATED        - #012->#005 K007 -  @"timer-b..y-event"[K005]  in <process "K001"[K003]> due 2022-06-24T06:58:06.664
C PROC_INST          ACTIVATE       - #013->#005 K008 - SERVICE_TASK "task" in <process "process"[K003]>
E PROC_INST_CREATION CREATED        - #014->#005 K009 - new <process "process"> with variables: {key=key-3515601627347623276}
C MSG_SUB            CREATE         - #015->  -1   -1 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E PROC_INST          ACTIVATING     - #016->#013 K008 - SERVICE_TASK "task" in <process "process"[K003]>
E JOB                CREATED        - #017->#013 K010 - K010 "task" @"task"[K008] 3 retries, in <process "process"[K003]> (no vars)
E PROC_INST          ACTIVATED      - #018->#013 K008 - SERVICE_TASK "task" in <process "process"[K003]>
E MSG_SUB            CREATED        - #019->#015 K011 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
C PROC_MSG_SUB       CREATE         - #020->  -1   -1 - "message-shouldSubscribeToScopeEvents" @[K005] in <process ?[K003]> (no vars)
E PROC_MSG_SUB       CREATED        - #021->#020 K006 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
C MSG                PUBLISH        - #022->  -1   -1 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 (no vars)
E MSG                PUBLISHED      - #023->#022 K012 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 (no vars)
E MSG_SUB            CORRELATING    - #024->#022 K011 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
C TIMER              TRIGGER        - #025->  -1 K007 -  @"timer-b..y-event"[K005]  in <process "K001"[K003]> due 2022-06-24T06:58:06.664
C PROC_MSG_SUB       CORRELATE      - #026->  -1   -1 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E TIMER              TRIGGERED      - #027->#025 K007 -  @"timer-b..y-event"[K005]  in <process "K001"[K003]> due 2022-06-24T06:58:06.664
E PROC_EVNT          TRIGGERING     - #028->#025 K013 -  @"timer-b..y-event"[K005] in <process K001[K003]> (no vars)
E PROC_EVNT          TRIGGERED      - #029->#025 K013 -  @"timer-b..y-event"[K005] in <process K001[K003]> (no vars)
E PROC_INST          ACTIVATING     - #030->#025 K014 - BOUNDARY_EVENT "timer-b..y-event" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #031->#025 K014 - BOUNDARY_EVENT "timer-b..y-event" in <process "process"[K003]>
C PROC_INST          COMPLETE       - #032->#025 K014 - BOUNDARY_EVENT "timer-b..y-event" in <process "process"[K003]>
E TIMER              CREATED        - #033->#025 K015 -  @"timer-b..y-event"[K005]  in <process "K001"[K003]> due 2022-06-24T07:58:06.668
C MSG                EXPIRE         - #034->  -1 K012 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 (no vars)
E PROC_MSG_SUB       CORRELATED     - #035->#026 K006 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E PROC_EVNT          TRIGGERING     - #036->#026 K016 -  @"message..y-event"[K005] in <process K001[K003]> (no vars)
E PROC_EVNT          TRIGGERED      - #037->#026 K016 -  @"message..y-event"[K005] in <process K001[K003]> (no vars)
E PROC_INST          ACTIVATING     - #038->#026 K017 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #039->#026 K017 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
C PROC_INST          COMPLETE       - #040->#026 K017 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
C MSG_SUB            CORRELATE      - #041->  -1   -1 - "message-shouldSubscribeToScopeEvents" (inter.) @[K005] in <process "process"[K003]> (no vars)
C JOB                COMPLETE       - #042->  -1 K010 - K010 "task" @""[  -1] -1 retries, in <process ?[?]> (no vars)
E PROC_INST          COMPLETING     - #043->#032 K014 - BOUNDARY_EVENT "timer-b..y-event" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #044->#032 K014 - BOUNDARY_EVENT "timer-b..y-event" in <process "process"[K003]>
E PROC_INST          SEQ_FLOW_TAKEN - #045->#032 K018 - SEQUENCE_FLOW "sequenc..69bafb7" in <process "process"[K003]>
C PROC_INST          ACTIVATE       - #046->#032 K019 - END_EVENT "endEven..1fe8fbb" in <process "process"[K003]>
C PROC_MSG_SUB       CORRELATE      - #047->  -1   -1 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E MSG                EXPIRED        - #048->#034 K012 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 (no vars)
E PROC_INST          COMPLETING     - #049->#040 K017 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #050->#040 K017 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
E PROC_INST          SEQ_FLOW_TAKEN - #051->#040 K020 - SEQUENCE_FLOW "sequenc..3ffd00b" in <process "process"[K003]>
C PROC_INST          ACTIVATE       - #052->#040 K021 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E MSG_SUB            CORRELATED     - #053->#041 K011 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E JOB                COMPLETED      - #054->#042 K010 - K010 "task" @"task"[K008] 3 retries, in <process "process"[K003]> (no vars)
E PROC_EVNT          TRIGGERING     - #055->#042 K022 -  @"task"[K008] in <process K001[K003]> (no vars)
C PROC_INST          COMPLETE       - #056->#042 K008 - SERVICE_TASK "task" in <process "process"[K003]>
E PROC_INST          ACTIVATING     - #057->#046 K019 - END_EVENT "endEven..1fe8fbb" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #058->#046 K019 - END_EVENT "endEven..1fe8fbb" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #059->#046 K019 - END_EVENT "endEven..1fe8fbb" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #060->#046 K019 - END_EVENT "endEven..1fe8fbb" in <process "process"[K003]>
E PROC_MSG_SUB       CORRELATED     - #061->#047 K006 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E PROC_EVNT          TRIGGERING     - #062->#047 K023 -  @"message..y-event"[K005] in <process K001[K003]> (no vars)
E PROC_EVNT          TRIGGERED      - #063->#047 K023 -  @"message..y-event"[K005] in <process K001[K003]> (no vars)
E PROC_INST          ACTIVATING     - #064->#047 K024 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #065->#047 K024 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
C PROC_INST          COMPLETE       - #066->#047 K024 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
C MSG_SUB            CORRELATE      - #067->  -1   -1 - "message-shouldSubscribeToScopeEvents" (inter.) @[K005] in <process "process"[K003]> (no vars)
E PROC_INST          ACTIVATING     - #068->#052 K021 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #069->#052 K021 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #070->#052 K021 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #071->#052 K021 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #072->#056 K008 - SERVICE_TASK "task" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #073->#056 K008 - SERVICE_TASK "task" in <process "process"[K003]>
E PROC_INST          SEQ_FLOW_TAKEN - #074->#056 K025 - SEQUENCE_FLOW "sequenc..3083c34" in <process "process"[K003]>
C PROC_INST          ACTIVATE       - #075->#056 K026 - END_EVENT "endEven..510a93d" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #076->#066 K024 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #077->#066 K024 - BOUNDARY_EVENT "message..y-event" in <process "process"[K003]>
E PROC_INST          SEQ_FLOW_TAKEN - #078->#066 K027 - SEQUENCE_FLOW "sequenc..3ffd00b" in <process "process"[K003]>
C PROC_INST          ACTIVATE       - #079->#066 K028 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E MSG_SUB            CORRELATED     - #080->#067 K011 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E PROC_INST          ACTIVATING     - #081->#075 K026 - END_EVENT "endEven..510a93d" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #082->#075 K026 - END_EVENT "endEven..510a93d" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #083->#075 K026 - END_EVENT "endEven..510a93d" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #084->#075 K026 - END_EVENT "endEven..510a93d" in <process "process"[K003]>
C PROC_INST          COMPLETE       - #085->#075 K005 - SUB_PROCESS "subprocess" in <process "process"[K003]>
E PROC_INST          ACTIVATING     - #086->#079 K028 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #087->#079 K028 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #088->#079 K028 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #089->#079 K028 - END_EVENT "endEven..001748c" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #090->#085 K005 - SUB_PROCESS "subprocess" in <process "process"[K003]>
C TIMER              CANCEL         - #091->#085 K015 -  @"timer-b..y-event"[K005]  in <process "K001"[K003]> due 2022-06-24T07:58:06.668
E PROC_MSG_SUB       DELETING       - #092->#085 K006 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
E PROC_INST          COMPLETED      - #093->#085 K005 - SUB_PROCESS "subprocess" in <process "process"[K003]>
E PROC_INST          SEQ_FLOW_TAKEN - #094->#085 K029 - SEQUENCE_FLOW "sequenc..9f83862" in <process "process"[K003]>
C PROC_INST          ACTIVATE       - #095->#085 K030 - END_EVENT "endEven..2253a04" in <process "process"[K003]>
C MSG_SUB            DELETE         - #096->  -1   -1 - "message-shouldSubscribeToScopeEvents" (inter.) @[K005] in <process ?[K003]> (no vars)
E TIMER              CANCELED       - #097->#091 K015 -  @"timer-b..y-event"[K005]  in <process "K001"[K003]> due 2022-06-24T07:58:06.668
E PROC_INST          ACTIVATING     - #098->#095 K030 - END_EVENT "endEven..2253a04" in <process "process"[K003]>
E PROC_INST          ACTIVATED      - #099->#095 K030 - END_EVENT "endEven..2253a04" in <process "process"[K003]>
E PROC_INST          COMPLETING     - #100->#095 K030 - END_EVENT "endEven..2253a04" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #101->#095 K030 - END_EVENT "endEven..2253a04" in <process "process"[K003]>
C PROC_INST          COMPLETE       - #102->#095 K003 - PROCESS "process" in <process "process"[K003]>
E MSG_SUB            DELETED        - #103->#096 K011 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
C PROC_MSG_SUB       DELETE         - #104->  -1   -1 - "message-shouldSubscribeToScopeEvents" (inter.) @[K005] in <process ?[K003]> (no vars)
E PROC_INST          COMPLETING     - #105->#102 K003 - PROCESS "process" in <process "process"[K003]>
E PROC_INST          COMPLETED      - #106->#102 K003 - PROCESS "process" in <process "process"[K003]>
E PROC_MSG_SUB       DELETED        - #107->#104 K006 - "message-shouldSubscribeToScopeEvents" correlationKey: key-3515601627347623276 @[K005] in <process "process"[K003]> (no vars)
```
